editor/src/components/canvas/grid-panels-state.tsx

```diff
@@ -189,23 +192,21 @@ export function updateLayout(
       const working = [...layout]
 
       // insert
-      working[update.columnIndex].panels = insert(
-        update.indexInColumn,
-        panelToInsert,
-        working[update.columnIndex].panels,
-      )
+      working[update.columnIndex] = {
+        ...working[update.columnIndex],
+        panels: insert(update.indexInColumn, panelToInsert, working[update.columnIndex].panels),
+      }
 
       return removeOldPanel(working)
     }
     if (update.type === 'after-index') {
       const working = [...layout]
 
       // insert
-      working[update.columnIndex].panels = insert(
-        update.indexInColumn + 1,
-        panelToInsert,
-        working[update.columnIndex].panels,
-      )
+      working[update.columnIndex] = {
+        ...working[update.columnIndex],
```

Comment:
Extracting out the existing value into a variable would probably make this a little cleaner.